Backend/router/v1/report.ctrl.js

```diff
@@ -47,11 +76,47 @@ const deleteReport = (req, res) => {
   }).then((result) => {
     res.status(204).json(result);  //no content
   }).catch((err) => {});
-}
+};
+
+const edit = (req, res) => {
+  const reportId = parseInt(req.params.reportId);
+  const newWhat = req.body.data[0].what;
+  const newLocation = req.body.data[0].location;
+  const newTime = req.body.data[0].time;
+  const newWho = req.body.data[0].who;
+  const newDetails = req.body.data[0].details;
+
+  if(req.isAuthenticated()){
+    models.report.update({
+        what: newWhat,
+        location: newLocation,
+        time: newTime,
+        who: newWho,
+        details: newDetails
+      },{
+        where: {id: reportId}
+    }).then((result) => {
+      res.status(200);
+      res.json({
+        msg: "update success"
+      });
+    }).catch((err) => {
+      res.json({
+        msg: "update fail"
+      });
+    });
+  } else {
+    res.status(401);
+    res.json({
+      msg: "you are not login"
```

Comment:
Use proper English in all debug messages. In fact, this message is a proper json response to the API (i.e. will be exposed to developers if we decide to open the API access), so it's more user-facing than debug messages. 

"You are not logged in."